### PR TITLE
Dh/dtm 549/cli testing and print support

### PR DESCRIPTION
**Pull request by rdhallman**
@hal-henningsmoen please review and merge.

### CODE REVIEW

File: lib/tasks/master.thor [hunk 1]
```diff
@@ -36,18 +50,56 @@ module Hodor
...
+       $ hodor master:exec :source_hive query.hql
+         - Pre-configured command that builds a command line from the "commands"
+       section of the clusters.yml file.
+
+       For example, if your clusters.yml file contains:
+
+       :test_cluster:
+           :commands:
+               :source_hive:
+                   :line: beeline -n hadoop -u jdbc:hive2://hadoop-cluster:10000/default -f
```
Comment (hal-henningsmoen) [inline lib/tasks/master.thor]: I'm not clear how/where the three following params below are processed.

**Comment by hal-henningsmoen** [comment]
Except for the one clarification I requested this LGTM.  These are useful enhancements.

**Comment by rdhallman** [comment]
@russellwhitaker primarily, these are the changes to Hodor that were required to allow the EMR setup Rakefiles to tap into Hodor's key/value pair system. A few other items included, like clearing execute permissons on files that didn't need that etc. Please review and commit.

**Comment by rdhallman** [comment]
Btw @russellwhitaker, @hal-henningsmoen already reviewed this PR, and my recent work is a response to his feedback, which required reverting and redoing some things. So, if you don't have time (given it is late in the day), I think we can consider this PR reviewed already. Up to you.

**Comment by russellwhitaker** [comment]
Works for me @dhallman

**Comment by hal-henningsmoen** [comment]
This looks great @dhallman.

**Pull request merged**